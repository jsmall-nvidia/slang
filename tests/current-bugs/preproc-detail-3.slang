//DIAGNOSTIC_TEST:SIMPLE:-E

// NOTE! This test should *fail*, if preprocessor is working correctly!

// Undefining a macro that is not defined within C/C++ is defined as *not* an error.
// On checking with DXC/FXC they also have this behavior.

#undef C