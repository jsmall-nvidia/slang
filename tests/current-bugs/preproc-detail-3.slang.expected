result code = 0
standard error = {
tests/current-bugs/preproc-detail-3.slang(14): warning 15401: macro 'C' is not defined
#undef C
       ^
}
standard output = {
Hey Something else 
}
